File: weather-master-frontend/src/types/weather.ts
export interface LocationRequest {
    location: string;
}

export interface WeatherResponse {
    location: string;
    current_temp: number;
    feels_like: number;
    condition: string;
    humidity: number;
    wind_speed: number;
    uv_index: number;
    timestamp: string;
}

export interface ForecastItem {
    date: string;
    temp_high: number;
    temp_low: number;
    condition: string;
    precipitation_chance: number;
}

export interface ClothingSuggestion {
    category: string;
    items: string[];
    reason: string;
}

export interface ActivitySuggestion {
    activity: string;
    description: string;
    suitability_score: number;
}

export interface CompleteSuggestions {
    weather: WeatherResponse;
    forecast: ForecastItem[];
    clothing: ClothingSuggestion[];
    activities: ActivitySuggestion[];
}

File: weather-master-backend/app/models/weather.py
from pydantic import BaseModel
from datetime import datetime
from typing import List

class LocationRequest(BaseModel):
    location: str  # city name or coordinates

class WeatherResponse(BaseModel):
    location: str
    current_temp: float
    feels_like: float
    condition: str  # sunny, rainy, cloudy, etc.
    humidity: int
    wind_speed: float
    uv_index: int
    timestamp: datetime

class ForecastItem(BaseModel):
    date: str
    temp_high: float
    temp_low: float
    condition: str
    precipitation_chance: int

class ClothingSuggestion(BaseModel):
    category: str  # outerwear, top, bottom, footwear, accessories
    items: List[str]
    reason: str

class ActivitySuggestion(BaseModel):
    activity: str
    description: str
    suitability_score: int  # 1-10

class CompleteSuggestions(BaseModel):
    weather: WeatherResponse
    forecast: List[ForecastItem]
    clothing: List[ClothingSuggestion]
    activities: List[ActivitySuggestion]
